### PR TITLE
Tested my project using Rspec functionality

### PR DESCRIPTION
**Pull request by MuhammadRaza29**

### CODE REVIEW

File: spec/jobs/scrapping_job_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require "rails_helper"
+
+RSpec.describe ScrappingJob, :type => :job do
+  describe "perform_later" do
+    it "having job enqueued" do
+      ActiveJob::Base.queue_adapter = :test
```
Comment (MuhammadRaza29) [inline spec/jobs/scrapping_job_spec.rb]: Kindly move it to one file which we use to set configurations of test environment.

File: spec/models/event_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe Event, type: :model do
+  subject {
+    Event.new(title: "Anything",
```
Comment (MuhammadRaza29) [inline spec/models/event_spec.rb]: kindly fix the alignment issue.

File: spec/models/event_spec.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    expect(subject).to_not be_valid
+  end
+
+  it "is valid without a date" do
+    subject.day = nil
+    expect(subject).to be_valid
+  end
+
+  it "is valid with two websource" do
+    subject.websource = ["berlinevent","visitevent"]
```
Comment (MuhammadRaza29) [inline spec/models/event_spec.rb]: Any event can have only one websource.

File: spec/jobs/scrapping_job_spec.rb (hunk 1 above)
Comment (moizansari22) [reply spec/jobs/scrapping_job_spec.rb]: in 'specs/rails_helper' , i have set the enviornment variable as ActiveJob::Base.queue_adapter = :test .
issue resolved .

File: spec/models/event_spec.rb (hunk 2 above)
Comment (moizansari22) [reply spec/models/event_spec.rb]: set the indentation as per requirements.

File: spec/models/event_spec.rb (hunk 3 above)
Comment (moizansari22) [reply spec/models/event_spec.rb]: solved . We assumed that websource should not be nil in any case.

File: app/controllers/events_controller.rb [hunk 4]
```diff
@@ -1,9 +1,9 @@
 class EventsController < ApplicationController
   def index
     if params[:websource].present?
-      @events = Event.where("websource LIKE ?", params[:websource]).paginate(:page => params[:page], :per_page => 10)
+      @events = Event.where("websource LIKE ?", params[:websource])
     else
-      @events = Event.all.paginate(:page => params[:page], :per_page => 10)
+      @events = Event.all
```
Comment (MuhammadRaza29) [inline app/controllers/events_controller.rb]: @moizansari22 Why you have removed pagination?

File: spec/models/event_spec.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe Event, type: :model do
+  subject {Event.new(title: "Anything",teaser: "Lorem ipsum",bar: "Panorama Bar",day: "Friday",date: "30.09.2022",websource: ["berlinevent","visitevent"])}
```
Comment (MuhammadRaza29) [inline spec/models/event_spec.rb]: @moizansari22 One main spec is missing.You need to create event object with valid values and check the if event object is created in DB or not.

File: spec/rails_helper.rb [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+# directory. Alternatively, in the individual `*_spec.rb` files, manually
+# require only the support files necessary.
+#
+# Dir[Rails.root.join('spec', 'support', '**', '*.rb')].sort.each { |f| require f }
+
+# Checks for pending migrations and applies them before tests are run.
+# If you are not using ActiveRecord, you can remove these lines.
+begin
+  ActiveRecord::Migration.maintain_test_schema!
+  ActiveJob::Base.queue_adapter = :test
```
Comment (MuhammadRaza29) [inline spec/rails_helper.rb]: kindly move into `environments/test.rb`.

**Comment by MuhammadRaza29** [comment]
@moizansari22  Kindly add description and share screenshot after running all rspecs.

File: spec/rails_helper.rb (hunk 6 above)
Comment (moizansari22) [reply spec/rails_helper.rb]: resolved .

File: spec/models/event_spec.rb (hunk 5 above)
Comment (moizansari22) [reply spec/models/event_spec.rb]: resolved

File: app/controllers/events_controller.rb (hunk 4 above)
Comment (moizansari22) [reply app/controllers/events_controller.rb]: added will_paginate functionality and written its Rspec script as per requirments

**Comment by moizansari22** [comment]
Screenshot 2022-09-29 at 1 11 35 PM

Tested my project using Rspec .

- tested jobs working fine with 3 tests scenario
- tested models and implemented 7 test cases
- tested controller with status code and index controller rendering
- tested views with displaying the events
- tested routes with actions in controller .
- total 15 test cases implemented

File: Gemfile [hunk 7]
```diff
@@ -86,3 +86,10 @@ gem 'letter_opener'
 gem 'sidekiq-scheduler'
 gem 'redis-rails'
 gem 'rspec'
+group :development, :test do
+  gem 'rspec-rails'
+end
+gem 'rails-controller-testing'
+gem 'will_paginate'
+gem 'grape'
+gem 'grape_on_rails_routes'
```
Comment (MuhammadRaza29) [inline Gemfile]:
```suggestion
gem 'grape_on_rails_routes'

```

File: app/api/event_store/base.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+module EventStore
+    class Base < Grape::API
+      mount EventStore::V2::Events
+    end
+end
```
Comment (MuhammadRaza29) [inline app/api/event_store/base.rb]:
```suggestion
end

```

File: app/controllers/api/v1/events_controller.rb [hunk 9]
```diff
@@ -0,0 +1,6 @@
+class Api::V1::EventsController < ApplicationController
+  def index
+    @events = Event.all
+    render json: @events
+  end
+end
```
Comment (MuhammadRaza29) [inline app/controllers/api/v1/events_controller.rb]:
```suggestion
end

```

**Pull request merged**